<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="world_idx" default="0" />

  <node pkg="lflh" name="lflh" type="run_policy.py" output="screen" required="true"
    args="$(arg world_idx)">
  </node>

  <!-- Start the navigation stack -->
  <!-- The configuration files are loaded from lflh -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">

    <rosparam file="$(find lflh)/configs/params/costmap_global_params.yaml"
      command="load" ns="global_costmap" />
    <rosparam file="$(find lflh)/configs/params/costmap_local_params.yaml"
      command="load" ns="local_costmap" />

    <rosparam file="$(find lflh)/configs/params/odom_nav_params/global_costmap_params.yaml"
      command="load" />
    <rosparam file="$(find lflh)/configs/params/odom_nav_params/local_costmap_params.yaml"
      command="load" />
    
    <rosparam file="$(find lflh)/configs/params/base_global_planner_params.yaml"
      command="load" />
    <rosparam file="$(find lflh)/configs/params/base_local_planner_params.yaml"
      command="load" />
    <rosparam file="$(find lflh)/configs/params/move_base_params.yaml" command="load" />

    <param name="base_global_planner" type="string" value="navfn/NavfnROS" />
    <param name="base_local_planner" value="local_planner/LocalPlanner" />

    <remap from="odom" to="odometry/filtered" />
    <remap from="cmd_vel" to="move_base/cmd_vel" />

  </node>

</launch>
